### PR TITLE
Ageojo/fruit intents

### PR DESCRIPTION
**Pull request by ageojo**
fruits interpreter: 
inquiry vs. quotes vs. trades vs. corrections
testing intents only

### CODE REVIEW

File: examples/fruits/custom/intents.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+entities:
+	# should we always include both `entities` and `intents`?
```
Comment (tshastry) [inline examples/fruits/custom/intents.yaml]: No, I'd leave it out here.

Comment (tshastry) [inline examples/fruits/custom/intents.yaml]: No, I'd leave it out here.

File: examples/fruits/tests.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+test: Quote
+transcript: I'll buy 70 purple carrots for $10 each
+intent: quote
+
+test: Trade
+transcript: Confirming I'll sell you 20 oranges for $20
+intent: trade
+
+test: Correction
+transcript: I have to change my offer. I'll buy 50 red apples for $100
```
Comment (tshastry) [inline examples/fruits/tests.txt]: Missing an intent label below this test.

File: test_discovery.py [hunk 3]
```diff
@@ -160,39 +160,57 @@ def test_single_entity(entities, test_name, test_value):
     return (0, 0)
 
 
-def test_single_case(test):
+def test_single_case(test_dict, response_intent_dict):      #most_likely_intent
```
Comment (tshastry) [inline test_discovery.py]: Could you move this comment to the comment section below and add more explanation?

File: test_discovery.py [hunk 4]
```diff
@@ -160,39 +160,57 @@ def test_single_entity(entities, test_name, test_value):
...
+       total_errors: number of entities in test whose observed value (str) differs from expected
+
+       total_char_errors: sum of number of char that differ between observed and expected values for each entity
+
+       characters:
+    """
+    # print("======\nTesting: {}".format(test['test']))
+    # resp = submit_transcript(test['transcript'])
+    #
+    # # Check if a valid response was received
```
Comment (tshastry) [inline test_discovery.py]: How come we're removing this check?

Comment (ageojo) [reply test_discovery.py]: this was moved to `test_all` so the intent can be checked first and, if the label returned differs from the expected intent, then entity testing for that intent is skipped

File: test_discovery.py [hunk 5]
```diff
@@ -255,8 +301,8 @@ def fail_test(resp, message="", continued=False):
 class cleanText(object):
     """Mock up a module that is imported by entities so they can be imported and inspected."""
     @staticmethod
-    def text2int(wordList, spacer):
-        return wordList
+    def text2int(word_list):
+        return word_list
```
Comment (mgoldey) [inline test_discovery.py]: Is this intentional? You're changing the number of arguments here

Comment (ageojo) [reply test_discovery.py]: it isn't used

**Pull request merged**